src/subcommand/server/database.rs

```diff
@@ -267,4 +267,127 @@ impl Database {
         .await
         .map_err(|err| anyhow!(err))
     }
+
+    pub async fn get_account(&self, username: &str) -> Result<Account> {
```

Comment:
```suggestion
    pub async fn get_account(&self, username: &str) -> Result<Option<Account>> {
```

Maybe it makes sense to return a None value since you already do fetch_optional and then in the server function (I'll comment there as well) you can do .ok_or_not_found() to automatically return a 404.